app/scripts/modules/core/delivery/executionGroup/executionGroup.component.ts

```diff
@@ -0,0 +1,159 @@
+
+  constructor(public $scope: IScope,
+              private $timeout: ITimeoutService,
+              private $state: angular.ui.IStateService,
+              private $stateParams: IExecutionDetailsStateParams,
+              private $uibModal: IModalService,
+              private executionService: any,
+              private collapsibleSectionStateCache: any,
+              private ExecutionFilterModel: any,
+              private pipelineConfigService: PipelineConfigService) {}
+
+  public $onInit() {
+    this.deploymentAccounts = uniq(flatten<string>(this.group.executions.map((e: any) => e.deploymentTargets))).sort();
+
+    this.pipelineConfig = find(this.application.pipelineConfigs.data, { name: this.group.heading });
+    this.strategyConfig = find(this.application.strategyConfigs.data, { name: this.group.heading });
+
+    this.viewState = {
+      triggeringExecution: false,
+      open: this.isShowingDetails() || !this.collapsibleSectionStateCache.isSet(this.getSectionCacheKey()) || this.collapsibleSectionStateCache.isExpanded(this.getSectionCacheKey()),
+      poll: null,
+      canTriggerPipelineManually: this.pipelineConfig,
+      canConfigure: this.pipelineConfig || this.strategyConfig,
+      showAccounts: this.ExecutionFilterModel.sortFilter.groupBy === 'name',
+    };
+
+    this.$scope.$on('toggle-expansion', (_, expanded) => {
+      if (this.viewState.open !== expanded) {
+        this.toggle();
+      }
+    });
+
+    this.$scope.$on('$stateChangeSuccess', () => {
+      // If the heading is collapsed, but we've clicked on a link to an execution in this group, expand the group
+      if (this.isShowingDetails() && !this.viewState.open) {
+        this.toggle();
+      }
+    });
+  }
+
+  private showDetails(executionId: string) {
+    return executionId === this.$stateParams.executionId &&
+      this.$state.includes('**.execution.**');
+  }
+
+  private isShowingDetails() {
+    return this.group.executions
+          .map((execution: any) => execution.id)
+          .some((id: string) => this.showDetails(id));
+  }
+
+  public configure(id: string) {
+    if (!this.$state.current.name.includes('.executions.execution')) {
+      this.$state.go('^.pipelineConfig', { pipelineId: id });
+    } else {
+      this.$state.go('^.^.pipelineConfig', { pipelineId: id });
+    }
+  }
+
+  private hideDetails() {
+    this.$state.go('.^');
+  }
+
+  private getSectionCacheKey() {
+    return this.executionService.getSectionCacheKey(this.ExecutionFilterModel.sortFilter.groupBy, this.application.name, this.group.heading);
+  };
+
+  public toggle() {
+    this.viewState.open = !this.viewState.open;
+    if (this.isShowingDetails()) {
+      this.hideDetails();
+    }
+    this.collapsibleSectionStateCache.setExpanded(this.getSectionCacheKey(), this.viewState.open);
+  }
+
+  private startPipeline(command: any) {
```

Comment:
Maybe? There isn't a public one, and the signature that the function expects doesn't match the signature of the internal ones that already exist.